consensus/blocknode.go

```diff
@@ -44,12 +25,14 @@ type blockNode struct {
 
 	height BlockHeight
 	depth  Target // Cumulative weight of all parents.
-	target Target // Target for next block.
+	target Target // Target for next block, i.e. any child blockNodes.
 
-	// Below are the set of diffs for the block, and a bool indicating if the
-	// diffs have been generated yet or not. Diffs are computationally
-	// expensive to generate, and so will only be generated if at some point
-	// the block is a part of the longest fork.
+	// Diffs are computationally expensive to generate, so a lazy approach is
+	// taken wherein the diffs are only generated when needed. A boolean
+	// prevents duplicate work from being performed.
+	//
+	// Note that diffsGenerated == true iff the node is in the State's
+	// currentPath; this is because diffs must be generated to apply the node.
```

Comment:
note quite right. diffsGenerated == true iff the node has been in the currentPath at some point.

If it's currently in the current path, diffsGenerated == true, but it could also be true if it used to be, but isn't anymore due to a fork.